### DISCUSSION TITLE
I don't think Nushell is ready for v1.0

### DISCUSSION THREAD

**Discussion by rayanamal** [nushell/nushell#13688] [General]
Well, nobody asked my opinion, and I am no maintainer.

But I find that nushell is version 0.97 a bit weird. Because even though it's so close to v1.0, it seems to me Nushell is not at all ready to launch. Nufmt is not done. Are we going to launch a language without a built-in formatter? Nupm is not done. Are we going to launch without a modern package manager? There are 1.3k issues. And so on. While Nushell is light years ahead of everything else, I think there is a significant amount of  work to be done before the official v1.0 release.

As a contrary point,  launching early might be better, because the prevailing criticism of Nushell at this point is "*But it's not v1.0 yet!*". 

Also, at least to me, it's obvious that Nushell has a place in the future, I don't think it'll be significantly negatively affected long-term by a slightly incomplete version 1.0.

Just some thoughts. The maintainers would obviously know the status and readiness of the project better than me.

**Comment by sholderbach** [comment]
That is not exactly news. `0.99.0` will not automatically wrap over to `1.0.0`, we will decide when we are ready to promise the necessary stability, that a version `1.0` will be released.

To get to stability we need the dedicated work, sincere input and cooperation from the community that is contributing to Nushell.
You can:
- help out with the currently underresourced projects like `nupm`, `nufmt`
- contribute by fixing bugs that are dragging on the contributors.
- make sure if you open issues they actually take the context of other issues and design goals into account
- help with documentation or online resources so folks have an easier time jumping in

**Reply by rayanamal** [reply]
Alright, thank you very much. I'll try to help out whenever I have some free time on my hands.

**Comment by yebei199** [comment]
Clearly, there is a need for `nufmt`. Why doesn't Nu Shell have a similar tool like `nufmt`, given that Python has `ruff`, TypeScript has `biome`, and Rust has `rustfmt`?

**Reply by ghost** [reply]
nufmt does exist with exactly that name <https://github.com/nushell/nufmt>, but its not quite production ready.

> given that Python has ruff, TypeScript has biome, and Rust has rustfmt?

Nushell is a shell-language used by hobbyists and nowhere near production.  
Python, TypeScript, and Rust and 3 of the [biggest languages](https://spectrum.ieee.org/top-programming-languages-2024) in the world.  
The amount of development-resources and money can't even remotely be compared.

Also there isn't even a PEP-8 equivalent (formatting standard) and the language itself is actively changing.  
Another reason for no 3rd-party formatting tool is the absence of other tooling (there is no good method to get a AST (try to find one, which shows both `?` and internal-command names)).

**Reply by yebei199** [reply]
> nufmt does exist with exactly that name https://github.com/nushell/nufmt, but its not quite production ready.nufmt 确实存在 https://github.com/nushell/nufmt 这个名称，但它还没有完全准备好生产。
> 
> > given that Python has ruff, TypeScript has biome, and Rust has rustfmt?鉴于 Python 有 ruff，TypeScript 有 biome，而 Rust 有 rustfmt？
> 
> Nushell is a shell-language used by hobbyists and nowhere near production.Nushell 是业余爱好者使用的一种 shell 语言，远非生产语言。 Python, TypeScript, and Rust and 3 of the [biggest languages](https://spectrum.ieee.org/top-programming-languages-2024) in the world.Python、TypeScript 和 Rust 以及世界上最大的 3 [种语言](https://spectrum.ieee.org/top-programming-languages-2024)。 The amount of development-resources and money can't even remotely be compared.开发资源和资金的数量甚至无法相提并论。
> 
> Also there isn't even a PEP-8 equivalent (formatting standard) and the language itself is actively changing.此外，甚至没有 PEP-8 等效项（格式标准），并且语言本身正在积极变化。 Another reason for no 3rd-party formatting tool is the absence of other tooling (there is no good method to get a AST (try to find one, which shows both `?` and internal-command names)).没有第三方格式化工具的另一个原因是没有其他工具（没有获取 AST 的好方法（尝试找到一个，它同时显示 `？`和内部命令名称））。

I just want the indentation alignment feature; after all, I don’t want to manually align it.

**Reply by ghost** [reply]
you can try to create it with regex, but it will be buggy.  
the only way to properly align things is using a AST of some sort. and that's more or less missing.

somthing i use is [helix](https://helix-editor.com/). it gives you a usable indentation while writing the code and also has the `&` command, which aligns multiple lines (or subsections of lines)
